chain/cosmos/broadcaster.go

```diff
@@ -0,0 +1,170 @@
+package cosmos
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/strangelove-ventures/ibctest/broadcast"
+	"github.com/strangelove-ventures/ibctest/internal/dockerutil"
+)
+
+var _ broadcast.Broadcaster = &Broadcaster{}
+
+type Broadcaster struct {
+	// buf stores the output sdk.TxResponse when broadcast.Tx is invoked.
+	buf *bytes.Buffer
+	// keyrings is a mapping of keyrings which point to a temporary test directory. The contents
+	// of this directory are copied from the node container for the specific user.
+	keyrings map[broadcast.User]keyring.Keyring
+
+	// chain is a reference to the CosmosChain instance which will be the target of the messages.
+	chain *CosmosChain
+	// t is the testing.T for the current test.
+	t *testing.T
+
+	// factoryOptions is a slice of broadcast.FactoryOpt which enables arbitrary configuration of the tx.Factory.
+	factoryOptions []broadcast.FactoryOpt
+	// clientContextOptions is a slice of broadcast.ClientContextOpt which enables arbitrary configuration of the client.Context.
+	clientContextOptions []broadcast.ClientContextOpt
+}
+
+// NewBroadcaster returns a instance of Broadcaster which can be used with broadcast.Tx to
+// broadcast messages sdk messages.
+func NewBroadcaster(t *testing.T, chain *CosmosChain) *Broadcaster {
+	return &Broadcaster{
+		t:        t,
+		chain:    chain,
+		buf:      &bytes.Buffer{},
+		keyrings: map[broadcast.User]keyring.Keyring{},
+	}
+}
+
+// ConfigureFactoryOptions ensure the given configuration functions are run when calling GetFactory
+// after all default options have been applied.
+func (b *Broadcaster) ConfigureFactoryOptions(opts ...broadcast.FactoryOpt) {
+	b.factoryOptions = append(b.factoryOptions, opts...)
+}
+
+// ConfigureClientContextOptions ensure the given configuration functions are run when calling GetClientContext
+// after all default options have been applied.
+func (b *Broadcaster) ConfigureClientContextOptions(opts ...broadcast.ClientContextOpt) {
+	b.clientContextOptions = append(b.clientContextOptions, opts...)
+}
+
+// GetFactory returns an instance of tx.Factory that is configured with this Broadcaster's CosmosChain
+// and the provided user.
```

Comment:
Wouldn't hurt to mention `ConfigureFactoryOptions` in the godoc for this method.